packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -1327,4 +1331,69 @@ export class KubernetesClient {
   public getContextsState(): Map<string, ContextState> {
     return this.contextsState.getContextsState();
   }
+
+  async execIntoContainer(
+    podName: string,
+    containerName: string,
+    onStdOut: (data: Buffer) => void,
+    onStdErr: (data: Buffer) => void,
+    onClose: () => void,
+  ): Promise<{ onStdIn: (data: string) => void; onResize: (columns: number, rows: number) => void }> {
+    let telemetryOptions = {};
+    try {
+      const ns = this.getCurrentNamespace();
+      const connected = await this.checkConnection();
+      if (!ns || !connected) {
+        throw new Error('not connected or namespace is empty');
```

Comment:
we should have separate errors instead of saying "or"

like..

"no namespace"

and

"not connected"

Reply:
splited into two different checks